feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go

```diff
@@ -0,0 +1,586 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package setupservice is scoped only to be used for scripts in path
+// feature/security/gnsi/certz/tests/client_certificates
+// Do not use elsewhere.
+package setupservice
+
+import (
+	context "context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	gnmipb "github.com/openconfig/gnmi/proto/gnmi"
+	spb "github.com/openconfig/gnoi/system"
+	authzpb "github.com/openconfig/gnsi/authz"
+	certzpb "github.com/openconfig/gnsi/certz"
+	gribipb "github.com/openconfig/gribi/v1/proto/service"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/knebind/creds"
+	p4rtpb "github.com/p4lang/p4runtime/go/p4/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	username = "certzuser"
+	password = "certzpasswd"
+	sn       = "role001.pop55.net.example.com"
+	servers  []string
+	retries  int
+)
+
+type rpcCredentials struct {
+	*creds.UserPass
+}
+
+func (r *rpcCredentials) GetRequestMetadata(_ context.Context, _ ...string) (map[string]string, error) {
+	return map[string]string{
+		"username": r.UserPass.Username,
+		"password": r.UserPass.Password,
+	}, nil
+}
+
+func (r *rpcCredentials) RequireTransportSecurity() bool {
+	return true
+}
+
+type entityType int8
+
+const (
+	// EntityTypeCertificateChain is type of entity of the certificate chain.
+	EntityTypeCertificateChain entityType = 0
+	// EntityTypeTrustBundle is type of entity of the trust bundle.
+	EntityTypeTrustBundle entityType = 1
+	// EntityTypeCRL is type of entity of the CRL.
+	EntityTypeCRL entityType = 2
+	// EntityTypeAuthPolicy is type of entity of the auth policy.
+	EntityTypeAuthPolicy entityType = 3
+)
+
+// CertificateChainRequest is an input argument for the  type definition for the  CreateCertzChain.
```

Comment:
since this struct definition is for the function after the function below, can you move it down there?

I think either all of this gets defined at the top of the file, or it gets defined 'right next to' the thing that requires it.
randomly in the middle of the file isn't helping readers and 'future you', because: "Wait, why is this here? is it important? <scroll, scroll, scroll... nope it's just randomly here? huh!>"